src/clr-angular/data/datagrid/built-in/filters/datagrid-numeric-filter.spec.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, ViewChild } from '@angular/core';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+
+import { TestContext } from '../../helpers.spec';
+import { ClrDatagridNumericFilterInterface } from '../../interfaces/numeric-filter.interface';
+import { CustomFilter } from '../../providers/custom-filter';
+import { FiltersProvider } from '../../providers/filters';
+import { Page } from '../../providers/page';
+import { StateDebouncer } from '../../providers/state-debouncer.provider';
+import { DomAdapter } from '../../../../utils/dom-adapter/dom-adapter';
+
+import { DatagridNumericFilter } from './datagrid-numeric-filter';
+import { DatagridNumericFilterImpl } from './datagrid-numeric-filter-impl';
+
+const PROVIDERS = [FiltersProvider, DomAdapter, Page, StateDebouncer];
+
+export default function(): void {
+  describe('DatagridNumericFilter component', function() {
+    // Until we can properly type "this"
+    let context: TestContext<DatagridNumericFilter<number>, FullTest>;
+    let filter: TestFilter;
+    let filtersInstance: FiltersProvider<number>;
+
+    function openFilter() {
+      context.clarityElement.querySelector('.datagrid-filter-toggle').click();
+      context.detectChanges();
+    }
+
+    beforeEach(function() {
+      filter = new TestFilter();
+      context = this.create(DatagridNumericFilter, FullTest, PROVIDERS);
+      filtersInstance = TestBed.get(FiltersProvider);
+    });
+
+    it('receives an input for the filter value', function() {
+      context.testComponent.filterValue = [null, 10];
+      context.detectChanges();
+      expect(context.clarityDirective.filter.value).toBe([null, 10]);
+    });
+
+    it('wires the RegisteredFilter correctly', function() {
+      const test = new DatagridNumericFilterImpl(new TestFilter());
+      context.testComponent.filter = filter;
+      context.detectChanges();
+      expect(context.clarityDirective.filter.filterFn).toEqual(test.filterFn);
+    });
+
+    it('receives an input for the filter logic', function() {
+      context.testComponent.filter = filter;
+      context.detectChanges();
+      expect(context.clarityDirective.filter.filterFn).toBe(filter);
+    });
+
+    it('registers a filter', function() {
+      context.clarityDirective.value = [null, 10];
+      expect(filtersInstance.getActiveFilters().length).toBe(1);
+      expect(filtersInstance.getActiveFilters()[0]).toBe(context.clarityDirective.filter);
+    });
+
+    it('registers itself as a CustomFilter provider', function() {
+      expect(context.testComponent.customFilter).toBe(context.clarityDirective);
+    });
+
+    // TODO: Should this count the inputs?
```

Comment:
I don't think that is necessary, we can use gemini to make a screenshot of the filter to make sure everything is looks correct.